src/kernel/syscalls.zig

```diff
@@ -0,0 +1,122 @@
+const std = @import("std");
+const scheduler = @import("scheduler.zig");
+const panic = @import("panic.zig").panic;
+const log = std.log.scoped(.syscalls);
+
+/// A compilation of all errors that syscall handlers could return.
+pub const Error = error{OutOfMemory};
```

Comment:
An idea:
As OutOfMemory is an `Allocator` thing, make `Error` a union of `Allocator.Error`. Then for `fromErrorCode` and `toErrorCode` use `error.OutOfMemory` (notice the lowercase 'e') or use the full `Allocator.Error.OutOfMemory` so if the Allocator interface changes, then the comptime code will fail so can update the error set

Reply:
So I considered making this an error union, but decided against it when I found out that errors of the same name can be coerced. For example, if error set `A` contains `a, b, c` and error set `B` contains `b, c`, you can return B.b from a function that returns A and it will be coerced (see [here](https://ziglang.org/documentation/master/#Errors)). Not using an error union here and instead using error coercion means that we only need to handle the errors that are actually thrown, rather than the whole set from the union. For example, if we add `d` to the `B` from above, but the functions that a syscall uses never actually return it, then we don't have to handle it here and add it to `A`, and we avoid unnecessary code. A more concrete example is that if e.g. the vfs couldn't return a `.NotAFile`, even if it is part of the vfs error set, then using an error union means we'd have to account for it anyway, but using error coercion means we wouldn't because it's never going to be returned!

Reply:
Fab, didn't consider the error coerction. I shall brush up on my error sets ;)